### PR TITLE
fix(api) check user rights and pageName validity in userFilter endpoints

### PR DESCRIPTION
**Pull request by a-launois**
## Description

Check user access rights on resource status page and pageName validity in the following endpoints:
```
centreon_application_user_getFilters        GET /{base_uri}api/{version}/users/filters/{pageName}                                                             
centreon_application_user_getFilter          GET /{base_uri}api/{version}/users/filters/{pageName}/{filterId}                                                  
centreon_application_user_addFilter         POST /{base_uri}api/{version}/users/filters/{pageName}                                                             
centreon_application_user_updateFilter    PUT /{base_uri}api/{version}/users/filters/{pageName}/{filterId}                                                  
centreon_application_user_patchFilter      PATCH /{base_uri}api/{version}/users/filters/{pageName}/{filterId}                                                  
centreon_application_user_deleteFilter     DELETE /{base_uri}api/{version}/users/filters/{pageName}/{filterId} 
```

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x
- [x] 24.04.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by paul-oureib — APPROVED** [review]

File: centreon/tests/api/features/UserFilter.feature [hunk 1]
```diff
@@ -10,6 +10,9 @@ Feature:
   Scenario: User filters
     Given I am logged in
 
+    When I send a GET request to '/api/v21.10/users/filters/unknown-page'
+    Then the response code should be "500"
```
Comment (wtermellil) [inline centreon/tests/api/features/UserFilter.feature:14]: Why not a 404 instead of 500 ?

Comment (a-launois) [reply centreon/tests/api/features/UserFilter.feature:14]: To me it doesn't qualify for a 404 error. The name 'pageName' is not very representative, I do not search an corresponding element in DB or anything. If I don't "search" I can't "found".

Comment (callapa) [reply centreon/tests/api/features/UserFilter.feature:14]: If the endpoint 'users/filters/unknown-page' is what you're looking for, a 404 error is preferable to a 500 error.

File: centreon/tests/api/features/UserFilter.feature [hunk 2]
```diff
@@ -10,6 +10,9 @@ Feature:
   Scenario: User filters
     Given I am logged in
 
+    When I send a GET request to '/api/v21.10/users/filters/unknown-page'
```
Comment (bdauria) [inline centreon/tests/api/features/UserFilter.feature:13]: we should write postman tests instead of behat as we are in the process of migrating them

**Review by arnapou — DISMISSED** [review]

**Review by jeremyjaouen — DISMISSED** [review]

**Review by paul-oureib — APPROVED** [review]

File: centreon/src/Centreon/Application/Controller/FilterController.php [hunk 3]
```diff
@@ -329,4 +348,31 @@ public function getFilter(string $pageName, int $filterId): View
...
+
+     /**
+      * @param Contact $user
+      *
+      * @throws \RestForbiddenException
+      */
+     private function userHasAccessToResourceStatusOrFail(Contact $user): void
+     {
+        if (! $user->hasTopologyRole(Contact::ROLE_MONITORING_RESOURCES_STATUS_RW)) {
+            throw new \RestForbiddenException(_('You are not allowed to access resources status'));
```
Comment (callapa) [inline centreon/src/Centreon/Application/Controller/FilterController.php:375]: Don't use classes from the "www" directory.

**Review by cg-tw — APPROVED** [review]

**Review by paul-oureib — APPROVED** [review]

**Review by dmyios — APPROVED** [review]

**Review by cg-tw — APPROVED** [review]

**Pull request merged**